tooltester/src/main/java/io/dockstore/tooltester/client/cli/Client.java

```diff
@@ -130,48 +204,318 @@ protected void setupClientEnvironment() {
+                    .mapToLong(tool -> tool.getVersions().parallelStream().filter(ToolVersion::getVerified).count());
+            System.out.println("Number of versions of tools to test on Dockstore (currently): " + longStream.sum());
+            toolTestResult = tools.parallelStream().filter(Tool::getVerified).map(this::testTool).reduce(true, Boolean::logicalAnd);
+            System.out.println("Successful \"testing\" of tools found on Dockstore: " + toolTestResult);
         } catch (ApiException e) {
             exceptionMessage(e, "", API_ERROR);
         }
-
         finalizeResults();
 
     }
 
     /**
+     * This function prints all the files from the verified tool
+     *
+     * @param verifiedTool The verified tool
+     */
+    public void printAllFilesFromTool(Tool verifiedTool) throws ApiException {
+        SourceFile dockerfile;
+        SourceFile descriptor;
+        SourceFile testParameter;
+        List<SourceFile> testParameterFiles;
+        DockstoreTool dockstoreTool;
+        Long containerId;
+        dockstoreTool = containersApi.getPublishedContainerByToolPath(verifiedTool.getId());
+        containerId = dockstoreTool.getId();
+        for (ToolVersion version : verifiedTool.getVersions()) {
+            String tag = version.getName();
+            dockerfile = containersApi.dockerfile(containerId, tag);
+            for (ToolVersion.DescriptorTypeEnum descriptorType : version.getDescriptorType()) {
+                switch (descriptorType.toString()) {
+                case "CWL":
+                    descriptor = containersApi.cwl(containerId, tag);
+                    break;
+                case "WDL":
+                    descriptor = containersApi.wdl(containerId, tag);
+                    break;
+                default:
+                    break;
+                }
+                testParameterFiles = containersApi.getTestParameterFiles(containerId, tag, descriptorType.toString());
+                for (SourceFile testParameterFile : testParameterFiles) {
+                    testParameter = testParameterFile;
+                    count++;
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates the pipeline(s) on Jenkins to test a tool
+     *
+     * @param tool The tool to create tests for
+     */
+    void createToolTests(Tool tool) {
+        DockstoreTool dockstoreTool = null;
+        try {
+            dockstoreTool = getContainersApi().getPublishedContainerByToolPath(tool.getId());
+        } catch (ApiException e) {
+            e.printStackTrace();
+        }
+        List<ToolVersion> toolVersions = tool.getVersions();
+        for (ToolVersion toolversion : toolVersions) {
+            Long containerId = dockstoreTool.getId();
+            String id = toolversion.getId();
+            String tag = toolversion.getName();
+            String name = id;
+            name = name.replace("/","-");
```

Comment:
I believe String format is mostly used to insert words into known locations of a String like:  String.format("Hello %s, %d", "world", 42);
In my case,  I'm simply replacing "/" with "-" where "/" may not even exist in the String.  I'm replacing "/" because Jenkins cannot create a job with "/" in the name.